### PR TITLE
File checksum field for Downloads.

### PR DESCRIPTION
**Pull request by mberacochea**
Import checksum files for AnalysisDownload files (**NOTE:** we need to define the name of the json files).

Also -> API performance improvements for:
- Analysis Downloads (from 116 queries to 26) on the download section.
- Other download sections to: Studies, Genome Releases and Genomes.

Added migration to create 3 algos MD5, SHA1 and SHA256

API response structure:
```json
{
    "type": "analysis-job-downloads",
    "id": "ERR2237853_MERGED_FASTQ_SSU_OTU.tsv",
    "attributes": {
    "alias": "ERR2237853_MERGED_FASTQ_SSU_OTU.tsv",
    "file-format": {
        "name": "TSV",
        "extension": "tsv",
        "compression": false
    },
    "description": {
        "label": "MAPseq SSU assignments",
        "description": "MAPSeq output file for SSU"
    },
    "group-type": "Taxonomic analysis SSU rRNA",
    "file-checksum": {
        "checksum": "065cb3c75d4d680a1f54b5ce7d5eb6b72ae84261",
        "checksum-algorithm": "SHA1"
    }
},
```

### CODE REVIEW

File: emgapi/migrations/0025_auto_20200615_1817.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('emgapi', '0024_auto_20200505_1737'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
```
Comment (mscheremetjew) [inline emgapi/migrations/0025_auto_20200615_1817.py:16]: Hey Martin, Out of interest, if you create a new table that way, will Django automatically add the new table to the Django model file?

File: emgapi/migrations/0026_auto_20200616_1202.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('emgapi', '0025_auto_20200615_1817'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_checksum_algorithms, reverse_code=remove_checksum_algorithms)
```
Comment (mscheremetjew) [inline emgapi/migrations/0026_auto_20200616_1202.py:42]: Sorry my ignorance, but I don't get the point here. Why do you add and remove the same checksum algorithms in one go?

File: emgapi/migrations/0025_auto_20200615_1817.py (hunk 1 above)
Comment (mberacochea) [reply emgapi/migrations/0025_auto_20200615_1817.py:16]: It's the other way. We added ```class ChecksumAlgorithm``` and Django created the migration for us.

**Comment by mberacochea** [comment]
> Hey Martin, overall it looks fine. The abstraction of a base download manager makes sense. Also Thanks for the implementation of the import_checksums.py tool.
> I do have a question though, with all these new model additions and the new import checksum tool, we also need to update the webuploader to populate these new checksum values. We will need to further discuss this. I have a vague idea how to integrate it, but would like to check with you beforehand.
> Also I realized that we are on migration number 26 on the webuploader branch as well, which means it will be even more fun to merge the 2 branch eventually.

Yes, we need to cherry pick the changes and fix the migrations. It shouldn't be difficult but merging both branches will be a pain, maybe we can put this merge on hold until we update the web VMs.

**Pull request merged**